interop-testing/src/test/java/io/grpc/testing/integration/CascadingTest.java

```diff
@@ -0,0 +1,334 @@
+      // The descendant RPCs are cancelled so they receive CANCELLED.
+      assertEquals(Status.Code.ABORTED, status.getCode());
+      // All nodes (15) except one edge of the tree (4) will be cancelled.
+      assertEquals(11, observedCancellations.get());
+      assertEquals(11, receivedCancellations.get());
+    }
+  }
+
+  /**
+   * Create a chain of client to server calls which can be cancelled top down.
+   */
+  private void startChainingServer() throws IOException {
+    final Executor otherWork = Context.propagate(Executors.newCachedThreadPool());
+    server = InProcessServerBuilder.forName("channel").addService(
+        ServerInterceptors.intercept(TestServiceGrpc.bindService(service),
+            new ServerInterceptor() {
+              @Override
+              public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(
+                  MethodDescriptor<ReqT, RespT> method,
+                  final ServerCall<RespT> call,
+                  Metadata headers,
+                  ServerCallHandler<ReqT, RespT> next) {
+                // Respond with the headers but nothing else.
+                call.sendHeaders(new Metadata());
+                call.request(1);
+                return new ServerCall.Listener<ReqT>() {
+                  @Override
+                  public void onMessage(final ReqT message) {
+                    try {
+                      // Wait and then recurse.
+                      if (depth.incrementAndGet() == depthThreshold) {
+                        call.close(Status.ABORTED, new Metadata());
+                        return;
+                      }
+                      Thread.sleep(50);
+                    } catch (InterruptedException ie) {
+                      throw new RuntimeException(ie);
+                    }
+
+                    otherWork.execute(new Runnable() {
+                      @Override
+                      public void run() {
+                        try {
+                          blockingStub.unaryCall((Messages.SimpleRequest) message);
+                        } catch (Throwable t) {
+                          Status status = Status.fromThrowable(t);
+                          if (status.getCode() == Status.Code.CANCELLED) {
+                            observedCancellations.incrementAndGet();
+                          } else if (status.getCode() == Status.Code.ABORTED) {
+                            call.close(status, new Metadata());
+                          }
+                        }
+                      }
+                    });
+                  }
+
+                  @Override
+                  public void onHalfClose() {
+                  }
+
+                  @Override
+                  public void onCancel() {
+                    receivedCancellations.incrementAndGet();
+                  }
+
+                  @Override
+                  public void onComplete() {
```

Comment:
No need to implement this. The default is NOOP. Ditto for onReady. Ditto for `startCallTreeServer`